### PR TITLE
[Federation] Convert ConfigMaps to use the generic sync controller

### PR DESCRIPTION
**Pull request by perotinus**
Integration tests pass. I'm having some issues getting the e2e tests up and running locally.

Directly addresses the issues in #44867.

cc @marun

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by cblecker** [comment]
@k8s-bot ok to test

**Comment by marun** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

File: test/e2e_federation/crud.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ var _ = framework.KubeDescribe("Federated types [Feature:Federation][Experimenta
 	f := fedframework.NewDefaultFederatedFramework("federated-types")
 
 	for name, fedType := range federatedtypes.FederatedTypes() {
-		Describe(fmt.Sprintf("Federated %q resources", name), func() {
+		Describe(fmt.Sprintf("dvorak Federated %q resources", name), func() {
```
Comment (marun) [inline test/e2e_federation/crud.go]: Heh

**Comment by marun** [comment]
I should hurry up and post tests for the sync controller.  It might make sense to add cluster addition to the crud test to cover a case that was covered by the removed unit test, and ensure that there is otherwise no loss in coverage.

**Comment by csbell** [comment]
@maru: By "post tests" you mean unit tests so that we recover the one that @perotinus removed in this PR? Thanks for clarifying, I was going to ask where the configmaps unit tests went.

File: test/e2e_federation/crud.go (hunk 1 above)
Comment (perotinus) [inline test/e2e_federation/crud.go]: :) Reverted this but forgot to repush.

**Comment by perotinus** [comment]
@csbell @marun I've restored the unit test. This should be ready to review now.

**Comment by nikhiljindal** [comment]
We dont have any e2e tests for configmap: https://github.com/kubernetes/kubernetes/pull/38384

**Comment by perotinus** [comment]
@nikhiljindal With this PR, we will: the test framework that @marun built has an e2e test that validates CRUD operations for types that have adapters.

**Comment by marun** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Comment by marun** [comment]
/lgtm

Just waiting on the results of the e2e test: http://prow.k8s.io/?type=presubmit&pull=44886&job=pull-kubernetes-federation-e2e-gce

**Comment by marun** [comment]
@csbell e2e is green, I think this is ready.

**Pull request merged**